From 1e6334b799ce776b14babf766a3ef58ecb15ab83 Mon Sep 17 00:00:00 2001
From: Shamim Ali <ashamim@ami.com>
Date: Thu, 17 Oct 2024 16:44:00 +0530
Subject: [PATCH] Add dts configuration for power operations

This patch used for gpio configuration for x86-power-control

gpioinfo:

root@evb-npcm845:~# gpioinfo 0
gpiochip0 - 32 lines:
        line   0:  "POWER_OUT"       unused  output  active-high
        line   1:   "PS_PWROK" "power-control" input active-high [used]
        line   2:  "RESET_OUT"       unused  output  active-high
        line   3: "POST_COMPLETE" "power-control" input active-high [used]
        line   4: "POWER_BUTTON" "power-control" input active-high [used]
        line   5: "RESET_BUTTON" "power-control" input active-high [used]
        line   6:    "NMI_OUT" "power-control" output active-high [used]
        line   7:      unnamed       unused  output  active-high

Test Cases:
root@evb-npcm845:~# ipmitool power off
Chassis Power Control: Down/Off
root@evb-npcm845:~# ipmitool power status
Chassis Power is on
root@evb-npcm845:~# ipmitool power status
Chassis Power is off
root@evb-npcm845:~#
root@evb-npcm845:~# ipmitool power on
Chassis Power Control: Up/On
root@evb-npcm845:~#
root@evb-npcm845:~# ipmitool power status
Chassis Power is on
root@evb-npcm845:~#

Signed-off-by: Shamim Ali <ashamim@ami.com>
---
 .../boot/dts/nuvoton/nuvoton-npcm845-evb.dts      | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
index 5154393f46fc..adb1ef2edfe5 100644
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
@@ -151,7 +151,7 @@ &tmps {
 &sgpio2 {
 	status = "okay";
 	gpio-line-names =
-		"POWER_OUT","RESET_OUT","","","","","","NMI_OUT",
+		"","","","","","","","",
 		"g_led","","","","","","","",
 		"","","","","","","","",
 		"","","","","","","","",
@@ -159,7 +159,7 @@ &sgpio2 {
 		"","","","","","","","",
 		"","","","","","","","",
 		"","","","","","","","",
-		"","","PS_PWROK","POST_COMPLETE","POWER_BUTTON","RESET_BUTTON","NMI_BUTTON","",
+		"","","","","","","NMI_BUTTON","",
 		"","","","","","","","",
 		"","","","","","","","",
 		"","","","","","","","",
@@ -169,6 +169,17 @@ &sgpio2 {
 		"","","","","","","","";
 };
 
+&pinctrl {
+        status = "okay";
+        gpio0: gpio@f0010000 {
+        gpio-line-names =
+            "POWER_OUT","PS_PWROK","RESET_OUT","POST_COMPLETE","POWER_BUTTON","RESET_BUTTON","NMI_OUT","",
+            "","","","","","","","",
+            "","","","","","","","",
+            "","","","","","","","";
+        };
+};
+
 &serial0 {
 	status = "okay";
 };
-- 
2.25.1

